src/Time/BoundaryHistory.hpp

```diff
@@ -103,6 +104,18 @@ class BoundaryHistory {
   size_t remote_size() const noexcept { return remote_data_.size(); }
   //@}
 
+  const LocalVars& local_data(const TimeStepId& time_id) const noexcept {
+    const Time& time = time_id.substep_time();
```

Comment:
This will have a very mild conflict with a PR I intend to submit soon, which changes the internal time storage in this class from `Time` to `TimeStepId`.  That change should actually make this function more robust.  It should be an easy merge for whichever of us needs to handle it.